src/doc/unstable-book/src/language-features/asm-experimental-arch.md

```diff
@@ -145,14 +151,19 @@ This feature tracks `asm!` and `global_asm!` support for the following architect
 | All          | `sp`, `r15` (s390x)                     | The stack pointer must be restored to its original value at the end of an asm code block.                                                                                           |
-| All          | `fr` (Hexagon), `$fp` (MIPS), `Y` (AVR), `r4` (MSP430), `a6` (M68k), `r11` (s390x), `x29` (Arm64EC) | The frame pointer cannot be used as an input or output.                                                                                                                             |
+| All          | `fr` (Hexagon), `fp` (PowerPC), `$fp` (MIPS), `Y` (AVR), `r4` (MSP430), `a6` (M68k), `r11` (s390x), `x29` (Arm64EC) | The frame pointer cannot be used as an input or output.                                                                                                                             |
 | All          | `r19` (Hexagon), `x19` (Arm64EC)        | This is used internally by LLVM as a "base pointer" for functions with complex stack frames.                                                                                        |
 | MIPS         | `$0` or `$zero`                         | This is a constant zero register which can't be modified.                                                                                                                           |
 | MIPS         | `$1` or `$at`                           | Reserved for assembler.                                                                                                                                                             |
 | MIPS         | `$26`/`$k0`, `$27`/`$k1`                | OS-reserved registers.                                                                                                                                                              |
 | MIPS         | `$28`/`$gp`                             | Global pointer cannot be used as inputs or outputs.                                                                                                                                 |
 | MIPS         | `$ra`                                   | Return address cannot be used as inputs or outputs.                                                                                                                                 |
 | Hexagon      | `lr`                                    | This is the link register which cannot be used as an input or output.                                                                                                               |
+| PowerPC      | `$r2`, `$r13`                           | These are system reserved registers.                                                                                                                                                |
```

Comment:
nit: Separate `r2` and `r13`.
`r2` is used as TOC pointer.
`r13` is system reserved register.

Reply:
Those explanations were just copied from the error message, but AFAIK r2 is used as a thread pointer in the PPC32 SVR4 ABI ([Linux](https://github.com/bminor/musl/blob/v1.2.5/arch/powerpc/pthread_arch.h), [FreeBSD](https://github.com/freebsd/freebsd-src/blob/release/14.1.0/sys/powerpc/include/tls.h#L52), [NetBSD](https://github.com/NetBSD/src/blob/432a1357026b10c184d8a0ddb683008a23cc7cd9/sys/arch/powerpc/include/mcontext.h#L159), [OpenBSD](https://github.com/openbsd/src/blob/ed8f5e8d82ace15e4cefca2c82941b15cb1a7830/sys/arch/powerpc/include/tcb.h#L37)), so I think the current explanation is actually fine as is.

https://github.com/rust-lang/rust/blob/d30a53e46962b7bd6bcc74e23cfb81c36e996eed/compiler/rustc_target/src/asm/powerpc.rs#L197-L198